src/interface/window.c

```diff
@@ -348,6 +349,46 @@ static void window_all_wheel_input()
 			return;
 }
 
+void window_close_surplus(int cap, rct_windowclass avoid_classification)
+{
+	int count, i, diff;
+	//find the amount of windows that are currently open
+	for (i = 0; i < MAX_WINDOW_COUNT; i++) {
+		if (&g_window_list[i] == RCT2_NEW_WINDOW) {
+			count = i;
+			break;
+		}
+	}
+	//difference between amount open and cap = amount to close
+	diff = count - cap; 
+	for (i = 0; i < diff; i++) {
+		rct_window *w = NULL;
+		//iterates through the list until it finds the newest window, or a window that can be closed
+		for (w = g_window_list; w < RCT2_NEW_WINDOW; w++) {
+			if (!(w->flags & (WF_STICK_TO_BACK | WF_STICK_TO_FRONT | WF_NO_AUTO_CLOSE)))
+				break;
+		}
+		//skip window if window is options (as user may be modifying via options)
+		if (avoid_classification != -1 && w != NULL && w->classification == avoid_classification) {
```

Comment:
I have since fixed this. Thank you for pointing this out. I decided, to ease rebasing and keeping this up to date, to merge all my commits together as well. Previously, I had to fix conflicts 3 or 4 separate times involving the string ids, now when I rebase I can just fix the merge conflict once and not need to worry about fixing it multiple times. But this is fixed in my newly rebased commit.